app/grandchallenge/evaluation/admin.py

```diff
@@ -58,6 +65,84 @@ def clean(self):
         return cleaned_data
 
 
+class ConfigureAlgorithmPhasesForm(Form):
+    phases = ModelMultipleChoiceField(
+        queryset=Phase.objects.select_related("challenge")
+        .filter(submission_kind=SubmissionKindChoices.CSV)
+        .all()
+    )
+    algorithm_inputs = ModelMultipleChoiceField(
+        queryset=ComponentInterface.objects.all()
+    )
+    algorithm_outputs = ModelMultipleChoiceField(
+        queryset=ComponentInterface.objects.all()
+    )
+
+    def clean(self):
+        cleaned_data = super().clean()
+        try:
+            for phase in self.cleaned_data["phases"]:
+                if Archive.objects.filter(
+                    slug=f"{slugify(phase.challenge.short_name)}-{slugify(phase.title)}-dataset"
+                ).exists():
+                    raise ValidationError(
+                        f"Archive for {phase} already exists."
+                    )
+        except KeyError:
+            pass
+        return cleaned_data
+
+
+class ConfigureAlgorithmPhasesView(PermissionRequiredMixin, FormView):
+    form_class = ConfigureAlgorithmPhasesForm
+    permission_required = "evaluation.configure_algorithm_phase"
+    template_name = "admin/evaluation/phase/algorithm_phase_form.html"
+    raise_exception = True
+
+    def form_valid(self, form):
+        for phase in form.cleaned_data["phases"]:
+            self.turn_phase_into_algorithm_phase(
+                phase=phase,
+                inputs=form.cleaned_data["algorithm_inputs"],
+                outputs=form.cleaned_data["algorithm_outputs"],
+            )
+        messages.success(self.request, "Phases were successfully updated")
+        return super().form_valid(form)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context.update(
+            {
+                **admin.site.each_context(self.request),
+                "opts": Phase._meta,
+                "title": "Turn phase(s) into algorithm phase(s)",
+            }
+        )
+        return context
+
+    def get_success_url(self):
+        return reverse("admin:evaluation_phase_changelist")
+
+    def turn_phase_into_algorithm_phase(self, *, phase, inputs, outputs):
+        archive = Archive.objects.create(
```

Comment:
The archive is missing a `full_clean` (I think that it's not valid anyway as it requires a logo).